apps/gateway/src/middlewares/models/gemini.ts

```diff
@@ -1,14 +1,96 @@
+import { BadRequestError } from "@hebo/shared-api/errors";
 
+import type { OpenAICompatibleReasoning } from "~gateway/utils/openai-compatible-api-schemas";
 
 import { ModelAdapterBase } from "./model";
 
+import type { ProviderOptions } from "@ai-sdk/provider-utils";
 
 export abstract class GeminiModelAdapter extends ModelAdapterBase {
   readonly modality = "chat";
   readonly owned_by = "google";
   readonly pricing = {
     monthly_free_tokens: 0,
   };
+
+  transformOptions(options?: ProviderOptions): ProviderOptions {
+    const modelConfig: Record<string, any> = {};
```

Comment:
Nitpick: any specific reason we use "config" for model vs "options" for provider? If no specific reason, let's streamline. Suggest Options, as modelConfig has overlapping meaning in the same repo.

(same comment applies to gpt.ts)